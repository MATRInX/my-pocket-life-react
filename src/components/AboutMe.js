import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { LINKS } from '../fixtures/links';
import selectAboutMePage from '../selectors/aboutMePage';

export const AboutMe = ({ aboutMePage }) => (
  <div>
    {
      aboutMePage === undefined ? (
        <div>Loading...</div>
      ) : (
        <div>
          <h2>Kilka słów o mnie</h2>
          <Link to={LINKS.ABOUT_ME}>
            <img src={aboutMePage.image}/>
          </Link>
          <p>
            Mam na imię Aneta i od półtora roku jestem mamą. Wraz z mężem i synem mieszkamy w Tarnowskich Górach.
          </p>
          <Link to={LINKS.ABOUT_ME}>Czytaj więcej...</Link>
        </div>
      )
    }    
  </div>
);

const mapStateToProps = ({ pages } = state) => ({
  aboutMePage: selectAboutMePage(pages)
});

export default connect(mapStateToProps)(AboutMe);