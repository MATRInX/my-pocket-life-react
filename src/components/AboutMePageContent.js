import React from 'react';

const AboutMePageContent = () => (
  <div>
    AboutMePageContent content...
  </div>
);

export default AboutMePageContent;