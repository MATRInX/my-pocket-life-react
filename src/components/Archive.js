import React from 'react';
import { connect } from 'react-redux';
import selectArchiveCategory from '../selectors/archiveCategory';
import { setArchiveMonth, setArchiveYear } from '../actions/filters';
import { history } from '../routers/AppRouter';

export class Archive extends React.Component {
  state = {
    selectValue: this.props.selectValue ? this.props.selectValue : 
      this.props.filters.archiveMonth + '|' + this.props.filters.archiveYear
  };

  componentDidMount() {
    if (this.state.selectValue !== '0|0') {
      const filters = this.state.selectValue.split('|');
      this.props.setArchiveYear(Number(filters[1]));
      this.props.setArchiveMonth(Number(filters[0]));
    }
  }

  onSelectChange = e => {
    const selectValue = e.target.value;
    this.setState(() => ({ selectValue }));
    const filters = selectValue.split('|');
    this.props.setArchiveYear(Number(filters[1]));
    this.props.setArchiveMonth(Number(filters[0]));
    history.push(`/archiwum/${filters[1]}/${filters[0]}`);
  }

  render() {
    return (
      <div>
        <p>Archive content: {this.state.selectValue}</p>
        <select 
          value={this.state.selectValue}
          onChange={this.onSelectChange}
        >
          <option value="0|0">Wybierz miesiÄ…c...</option>
          {
            this.props.archiveData.map((archiveItem, index) => {
              return (
                <option 
                  key={index} 
                  value={archiveItem.month+'|'+archiveItem.year}
                >
                  {archiveItem.text}  
                </option>
              )
            })
          }
        </select>
      </div>
    );
  }
};

const mapStateToProps = ({ posts, filters } = state) => ({
  archiveData: selectArchiveCategory(posts),
  filters
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  setArchiveMonth: month => dispatch(setArchiveMonth(month)),
  setArchiveYear: year => dispatch(setArchiveYear(year))
});

export default connect(mapStateToProps, mapDispatchToProps)(Archive);