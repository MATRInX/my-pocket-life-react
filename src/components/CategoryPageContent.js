import React from 'react';

const CategoryPageContent = (props) => (
  <div>
    CategoryPageContent - category:
    <p>{props.match.params.category}</p>
  </div>
);

export default CategoryPageContent;