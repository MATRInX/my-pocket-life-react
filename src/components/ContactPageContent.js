import React from 'react';

const ContactPageContent = () => (
  <div>
    ContactPageContent content...
  </div>
);

export default ContactPageContent;