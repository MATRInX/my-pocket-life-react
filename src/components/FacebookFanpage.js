import React from 'react';

const FacebookFanpage = () => (
  <div>FacebookFanpage content</div>
);

export default FacebookFanpage;