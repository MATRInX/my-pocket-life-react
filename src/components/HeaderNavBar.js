import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import LINKS from '../fixtures/links';

const HeaderNavBar = () => (
  <nav>
    <ul>
      <li>
        <a href={LINKS.HOME}>
          <FontAwesomeIcon icon="home" />
        </a>
      </li>
      <li>
        <a href={LINKS.ABOUT_ME}>O mnie</a>
      </li>
      <li>
        <a href={LINKS.CONTACT}>Kontakt</a>
      </li>
      <li>
        <a href={LINKS.FACEBOOK} target="_blank">
          <FontAwesomeIcon icon={['fab', 'facebook-f']} />
        </a>
      </li>
      <li>
        <a href={LINKS.INSTAGRAM} target="_blank">
          <FontAwesomeIcon icon={['fab', 'instagram']} />
        </a>
      </li>
    </ul>
  </nav>
)

export default HeaderNavBar;  
