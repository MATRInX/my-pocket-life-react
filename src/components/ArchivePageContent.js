import React from 'react';
import Archive from './Archive';
import PostList from './PostsList';

const ArchivePageContent = ({ match }) => (
  <div>
    ArchivePageContent content...
    <PostList />
    <Archive selectValue={match.params.month +'|'+match.params.year}/>
  </div>
);

export default ArchivePageContent;