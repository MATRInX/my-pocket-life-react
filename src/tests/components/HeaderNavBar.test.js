import React from 'react';
import { shallow } from 'enzyme';
import { HeaderNavBar } from '../../components/HeaderNavBar';

test('should correctly render HeaderNavBar', () => {
  const wrapper = shallow(<HeaderNavBar />);
  expect(wrapper).toMatchSnapshot();
});