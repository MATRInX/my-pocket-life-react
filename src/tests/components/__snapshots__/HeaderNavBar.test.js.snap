// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render HeaderNavBar 1`] = `
<nav>
  <ul>
    <li>
      <Link
        replace={false}
        to="/"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="home"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/o-mnie"
      >
        O mnie
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/kontakt"
      >
        Kontakt
      </Link>
    </li>
    <li>
      <a
        href="https://www.facebook.com/MyPocketLife/"
        target="_blank"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fab",
              "facebook-f",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </a>
    </li>
    <li>
      <a
        href="https://www.instagram.com/aneta.pos/"
        target="_blank"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fab",
              "instagram",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </a>
    </li>
  </ul>
</nav>
`;
